### PR TITLE
Fragmentation

### PR DESCRIPTION
**Pull request by SteffenHeu**
fragment graph for ms2 spectra

### CODE REVIEW

**Review by robinschmid — CHANGES_REQUESTED** [review]
Code looks great and the dashboard is amazing. 
I mostly found issues with the TableColumns. JavaFx tables and columns are quite complex so it might be better to use the factories of TableColumns and create new ones so that it becomes easier in the future.  
Currently the ColumnValue types are sometimes String sometimes Double or just the TableView Row type again. 

I pushed minor changes that came up in the review. Mostly finals and npe checks.

Also created an issue to track other future ideas that will not go into this PR. 
https://github.com/mzmine/mzmine/issues/2055

File: mzmine-community/src/main/java/io/github/mzmine/modules/tools/fraggraphdashboard/spectrumplottable/NeutralLossAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import org.jfree.chart.annotations.AbstractXYAnnotation;
+import org.jfree.chart.axis.ValueAxis;
+import org.jfree.chart.plot.Plot;
+import org.jfree.chart.plot.PlotOrientation;
+import org.jfree.chart.plot.PlotRenderingInfo;
+import org.jfree.chart.plot.XYPlot;
+import org.jfree.chart.ui.RectangleEdge;
+import org.jfree.chart.util.LineUtils;
+
+public class NeutralLossAnnotation extends AbstractXYAnnotation {
```
[the remark this answers is not in the document]
Comment (SteffenHeu) [reply mzmine-community/src/main/java/io/github/mzmine/modules/tools/fraggraphdashboard/spectrumplottable/NeutralLossAnnotation.java:36]: yes but will be uses

File: mzmine-community/src/main/java/io/github/mzmine/modules/tools/fraggraphdashboard/nodetable/EdgeTable.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    signal2.setComparator(Comparator.comparingDouble(this::mzDoubleParser));
+    signal2.setMinWidth(70);
+
+    TableColumn<SubFormulaEdge, String> formulaDifference = new TableColumn<>("Formula\ndiff.");
+    formulaDifference.setCellValueFactory(
+        cell -> cell.getValue().lossFormulaStringProperty().map(str -> STR."-[\{str}]"));
+    formulaDifference.setMinWidth(85);
+
+    TableColumn<SubFormulaEdge, String> massDifferenceAbs = new TableColumn<>(
+        "Mass diff.\n(meas.)");
```
[the remark this answers is not in the document]
Comment (SteffenHeu) [reply mzmine-community/src/main/java/io/github/mzmine/modules/tools/fraggraphdashboard/nodetable/EdgeTable.java:84]: measured

**Review by robinschmid — APPROVED** [review]
Great change with the TableColumns factories. It is much more readable now and easy to understand the structure of the tables. 

Can be merged. See my PR to your branch that harmonized the factories in TableColumns and added the Comparator as option so that mass difference columns can be created with one line.

As Property will create a TableColumn<T, Double> whereas DoubleProperty creates a TableColumn<T, Number> but thats ok.

**Pull request merged**